src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2500,6 +2500,9 @@ std::string Creature::GetScriptName() const
 
 uint32 Creature::GetScriptId() const
 {
+    if (CreatureData const* creatureData = GetCreatureData())
```

Comment:
Check ObjectMgr changes.
If we can't find a valid script from the creature table, we check creature_template.